src/regmod/models/tobit.py

```diff
@@ -0,0 +1,238 @@
+"""
+Tobit Model
+"""
+# pylint: disable=C0103
+from functools import partial
+from typing import List, Optional
+
+from jax import grad, hessian, jit, lax
+from jax.numpy import DeviceArray
+import jax.numpy as jnp
+from jax.scipy.stats.norm import logcdf, logpdf
+from numpy.typing import ArrayLike
+from pandas import DataFrame
+from regmod.data import Data
+from regmod.function import SmoothFunction
+
+from .model import Model
+
+
+identity_jax = SmoothFunction(
+    name="identity_jax",
+    fun=jnp.array,
+    inv_fun=jnp.array,
+    dfun=jnp.ones_like,
+    d2fun=jnp.zeros_like
+)
+
+
+exp_jax = SmoothFunction(
+    name="exp_jax",
+    fun=jnp.exp,
+    inv_fun=jnp.log,
+    dfun=jnp.exp,
+    d2fun=jnp.exp
+)
+
+
+class TobitModel(Model):
+    """Tobit model class.
+
+    A tobit regression model is used for data generated by a Gaussian
+    distribution with all negative elements set to 0 (i.e., a censored
+    or rectified Gaussian distribution).
+
+    """
+
+    param_names = ("mu", "sigma")
+    default_param_specs = {
+        "mu": {"inv_link": identity_jax},
+        "sigma": {"inv_link": exp_jax}
+    }
+
+    def __init__(self, data: Data, **kwargs) -> None:
+        """Initialize tobit model.
+
+        Parameters
+        ----------
+        data : Data
+            Training data.
+
+        Raises
+        ------
+        ValueError
+            If negative observations present in `data`.
+
+        Notes
+        -----
+        User input for parameter attribute `inv_link` is ingored and
+        defaults used instead.
+
+        """
+        if not jnp.all(data.obs >= 0):
+            raise ValueError("Tobit model requires non-negative observations.")
+        super().__init__(data, **kwargs)
+
+        # Use JAX inv_link functions
+        for ii, param_name in enumerate(self.param_names):
+            default_link = self.default_param_specs[param_name]["inv_link"]
+            self.params[ii].inv_link = default_link
+
+        # Use JAX data structures
+        self.mat = [jnp.asarray(mat) for mat in self.mat]
+        self.uvec = jnp.asarray(self.uvec)
+        self.gvec = jnp.asarray(self.gvec)
+        self.linear_uvec = jnp.asarray(self.linear_uvec)
+        self.linear_gvec = jnp.asarray(self.linear_gvec)
+        self.linear_umat = jnp.asarray(self.linear_umat)
+        self.linear_gmat = jnp.asarray(self.linear_gmat)
+
+    @partial(jit, static_argnums=(0,))
+    def objective(self, coefs: ArrayLike) -> float:
+        """Get negative log likelihood wrt coefficients.
+
+        Parameters
+        ----------
+        coefs : array_like
+            Model coefficients.
+
+        Returns
+        -------
+        float
+            Negative log likelihood.
+
+        """
+        weights = self.data.weights*self.data.trim_weights
+        obj_param = weights*self.get_nll_terms(coefs)
+        return obj_param.sum() + self.objective_from_gprior(coefs)
```

Comment:
`get_nll_terms` have the data weights, here we only need to apply the trim weights